sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/BeamSqlEnv.java

```diff
@@ -134,9 +141,21 @@ public void executeDdl(String sqlStatement) throws ParseException {
 
   public String explain(String sqlString) throws ParseException {
     try {
-      return RelOptUtil.toString(planner.convertToBeamRel(sqlString));
+      return RelOptUtil.toString(sqlEngine.convertToBeamRel(sqlString));
     } catch (ValidationException | RelConversionException | SqlParseException e) {
       throw new ParseException("Unable to parse statement", e);
     }
   }
+
+  public void setOptions(BeamSqlPipelineOptions options) {
+    this.options = options;
+  }
+
+  private BeamSqlPipelineOptions getBeamSqlPipelineOptions() {
+    if (options == null) {
+      options = PipelineOptionsFactory.create().as(BeamSqlPipelineOptions.class);
```

Comment:
Will this create the empty options? There's an existing `getPipelineOptions()` method here, why not use the same approach? I would actually rename `Map<> getPipelineOptions()` to `getPipelineOptionsMap()` and implement the new `BeamSqlBeamPipelineOptions getPipelineOptions()`

Reply:
I will rename `getPipelineOptions` to `getPipelineOptionsMap` or something else. 

The naming of `getPipelineOptions` is confusing. It does not get the pipeline options we want. This is a function used in `BeamEnumerableConverter` to create a pipeline options in the JDBC path.

Reply:
Actually this function is only being used in `BeamSqlCli`

Reply:
Why cannot we get the pipelineoptions from schema here?

Reply:
Because the pipeline options from schema is [empty](https://github.com/apache/beam/blob/master/sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/JdbcDriver.java#L181). It is not initialized by "[PipelineOptions.java](https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptions.java)".

Reply:
The PipelineOptions from the schema is easily accessible but isn't useful for this purpose, as it isn't the same as the PipelineOptions used in SqlTransform. In BeamSqlCli and JdbcDriver, PipelineOptions can't be used because we need a parser to configure pipeline options. This goes back to my comment from Friday: BeamSqlEnv should have an interface taking either a String classpath or Constructor function for the planner.